code/__HELPERS/maths.dm

```diff
@@ -94,13 +94,19 @@
 		return "[round(units * 0.000001, 0.001)] MJ"
 	return "[round(units * 0.000000001, 0.0001)] GJ"
 
+/proc/joules_to_energy(joules)
+	return joules * (1 SECONDS) / SSmachines.wait
+
+/proc/energy_to_joules(energy_units)
+	return energy_units * SSmachines.wait / (1 SECONDS)
+
 /// Format an energy value measured in Power Cell units.
 /proc/display_energy(units)
 	// APCs process every (SSmachines.wait * 0.1) seconds, and turn 1 W of
 	// excess power into GLOB.CELLRATE energy units when charging cells.
 	// With the current configuration of wait=20 and CELLRATE=0.002, this
 	// means that one unit is 1 kJ.
-	return display_joules(units * SSmachines.wait * 0.1 / GLOB.CELLRATE)
+	return display_joules(energy_to_joules(units) WATTS)
```

Comment:
With how often this is likely to fire, making the proc here a macro or just directly putting the code in will likely help with performance.

Reply:
The `display_energy` proc only fires on examines.